core/models.py

```diff
@@ -0,0 +1,125 @@
+"""Product, ProductReview, Category, Cart, Order, OrderItem Model
+
+Models are inherited from the Django.db.models.Model. These models are
+used to create, update, delete a new product. We can also add a product to
+cart. We can place an order.
+"""
+
+import datetime
+from django.db import models
+from authentication.models import User
+
+
+class Category(models.Model):
+    """Category Model
+    This is the category model for the authentication app. The category model
+    is used to map the category to the database. This model is used to create,
+    update and delete a category.
+    """
+
+    name = models.CharField(max_length=50, unique=True, blank=False)
+
+    # pylint: disable=too-few-public-methods
+    class Meta:
+        """Meta class"""
+        ordering = ["name"]
+
+
+class Product(models.Model):
+    """Product Model
+
+    This is the product model for the authentication app. The product model is
+    used to map the product to the database. This model is used to create,
+    update and delete a product. In this model we used the foreign key to map
+    the category to the product.
+    In this model we used the fields stock and sold to keep track of the stock.
+
+    """
+
+    name = models.CharField(max_length=60, unique=True, blank=False)
+    price = models.IntegerField(default=0)
```

Comment:
since price value can not be less than 0, so we can also add validator here, something like

```
from django.core.validators import MinValueValidator
price = models.IntegerField(validators=[MinValueValidator(0)], default=0)
```

We can add this min or max validator for other fields too (wherever required)